src/page.ts

```diff
@@ -39,6 +43,11 @@ export type WaitForNetworkIdleOptions = {
   idleConnections?: number;
 };
 
+export type SandboxOptions = {
+  sandbox?: boolean | Network_ResourceType[];
```

Comment:
Is there a reason that a user would only want to sandbox certain types of request? What's the usecase here? Wouldn't this be a security foot-gun?

Reply:
I think it depends on whether you want granularity at "website level" or "asset level"

My use-case would be the former, where I'd just like to control what website can be accessed, but don't really care where they need to load assets from (like if they use external domains from CDN or hotlinks).
I guess it'd limit my checks to `Document` requests, as for me static assets like `Stylesheet`, `Image`, `Media`, `Font` or  `Script` would be fine as I'd consider I already "trusted" the document itself.

But to be honest the [`Network.ResourceType`](https://chromedevtools.github.io/devtools-protocol/tot/Network/#type-ResourceType) has a lot of types (Document, Stylesheet, Image, Media, Font, Script, TextTrack, XHR, Fetch, Prefetch, EventSource, WebSocket, Manifest, SignedExchange, Ping, CSPViolationReport, Preflight, Other) and I'm not sure what all of them corresponds too (like `EventSource`, `SignedExchange`, `Preflight` or `Other`).

I feel like the current API I offer in this PR exposes a bit too much stuff, but I really think we should have something to address the "website level" or "asset level" control. 
I can see where both could be useful, like if you're testing your own website maybe you want to be sure that you load nothing suspicious, but if you're just webscraping stuff, maybe you just want to restrict which website can be accessed but you may not know in advance where the website will load its content

Reply:
Finally decided that it's better to just leave a boolean for `sandbox` to make it less complex.
If it's too limiting I guess it can be improved later on

Reply:
Agree on the decision. I think we could formulate a less leaky abstraction at some point in the future if a need is discovered.